### PR TITLE
Datastore read subclasses discrim only

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
related #1475 
read discrim only subclasses

### CODE REVIEW

File: spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java [hunk 1]
```diff
@@ -67,15 +67,24 @@ public void setApplicationContext(ApplicationContext applicationContext)
...
 	 */
-	public static void addDiscriminationClassConnection(Class classA, Class classB) {
-		Set<Class> setA = discriminationFamilies.computeIfAbsent(classA, DatastoreMappingContext::createSingletonSet);
-		Set<Class> setB = discriminationFamilies.computeIfAbsent(classB, DatastoreMappingContext::createSingletonSet);
-
-		setA.addAll(setB);
-		discriminationFamilies.put(classB, setA);
+	public static void addDiscriminationClassConnection(Class parentClass, Class subClass) {
+		Set<Class> setParent = discriminationFamilies.computeIfAbsent(parentClass,
+				unused -> createSingletonSet(subClass));
```
Comment (dmitry-s) [inline spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java]: if you add subClass later anyway, just return an empty set here

Comment (ChengyuanZhao) [reply spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java]: Done.

File: spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java [hunk 2]
```diff
@@ -67,15 +67,23 @@ public void setApplicationContext(ApplicationContext applicationContext)
...
+				addDiscriminationClassConnection(parentClass, x);
+			}
+		});
+		Class grandParent = parentClass.getSuperclass();
+		if (grandParent != null) {
+			addDiscriminationClassConnection(grandParent, subClass);
+		}
 	}
 
 	private static Set<Class> createSingletonSet(Class aClass) {
```
Comment (dmitry-s) [inline spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java]: remove `createSingletonSet` if it's unused

Comment (ChengyuanZhao) [reply spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java]: done

**Pull request merged**